src/pages/logs-page.js

```diff
@@ -90,7 +90,9 @@ export function LogsPage() {
   const [topologyId, setTopologyId] = useState(searchParams.get("topologyId"));
   const [externalId, setExternalId] = useState(searchParams.get("externalId"));
   const [type, setType] = useState(searchParams.get("type"));
-  const [start, setStart] = useState(timeRanges[0].value);
+  const [start, setStart] = useState(
+    searchParams.get("start") || timeRanges[0].value
+  );
```

Comment:
Get start directly from params, instead of setting another state?

Reply:
Didn't get that. Would there be an issue with something like:
```diff
modified   src/pages/logs-page.js
@@ -90,9 +90,8 @@ export function LogsPage() {
   const [topologyId, setTopologyId] = useState(searchParams.get("topologyId"));
   const [externalId, setExternalId] = useState(searchParams.get("externalId"));
   const [type, setType] = useState(searchParams.get("type"));
-  const [start, setStart] = useState(
-    searchParams.get("start") || timeRanges[0].value
-  );
+
+  const start = searchParams.get("start") || timeRanges[0].value;
 
   const [topologies, setTopologies] = useState([]);
   const [logs, setLogs] = useState([]);
@@ -223,7 +222,7 @@ export function LogsPage() {
             name="start"
             className="w-40 mr-2 flex-shrink-0"
             items={timeRanges}
-            onChange={(e) => setStart(e)}
+            onChange={(e) => setSearchParams({ start: e })}
             value={start}
           />
         </>
```

Reply:
@ciju I have url params syncing logic at one place. This code is already in place so followed the same pattern here. May I know why this approach is better compared to previous one.

Reply:
Ignore. Will setup a pattern to follow (and reasoning), later. You can close resolve this issues.